magento/pwa-studio
# Added new prototype feature to display social media icons with links

## Description
I added a new component which currently takes its configuration from a util-class getMediaConfig. I implemented this because at some point the social media icons should be configured in magento and loaded over graphql. The implementation need for that is hard dependent on 2120.

The SocialMedia Icons have a feature toggle which can be used to either render them in the footer or not. Currently there is no feature to only display for example facebook and youtube only. If the feature is enabled it will display all 4 social media icons.

I also added the four icons for the social media platforms twitter, facebook, instagram and youtube. These icons are from https://feathericons.com/.

For testing purpose I added some jest tests.
 
## Related Issue





Closes #2152 .

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps


1. Go to the any page.
2. Verify that by default social media icons are displayed like this:
<img width="1420" alt="social-media-icons" src="https://user-images.githubusercontent.com/46490366/74099500-c852d900-4b24-11ea-8bad-4da5238ff8ed.png">
3. Every Icon should link to the specific social media homepage

## Screenshots
### Mobile
<img width="359" alt="social-media-icons-mobile" src="https://user-images.githubusercontent.com/46490366/74099564-7f4f5480-4b25-11ea-81e7-f56dd2f90085.png">

## Things to improve
- Graphql Endpoint in Magento to get the configuration from the core_config_data table. (depends on #2120)
- Build in a check to render a social media icon only if a link is configured in magento (currently all four icons are rendered all the time. Except if you disable the feature)

## Checklist

* I have updated the documentation accordingly, if necessary. (not necessary)
* I have added tests to cover my changes, if necessary. (done)
